BookStore.hs

```diff
@@ -91,6 +95,21 @@ sampleCatalog =
       }
   ]
 
+booksByOldAuthors :: Int -> Int -> Catalog -> Catalog
+booksByOldAuthors maxYearOfBirth minYearOfPublication = filter (\book -> any (\author -> yearOfBirth author <= maxYearOfBirth) (authors book) && yearOfPublication book > minYearOfPublication)
```

Comment:
Could you try to compose two other functions in this file to achieve the same result?